features/nanostack/FEATURE_NANOSTACK/mbed-mesh-api/source/LoWPANNDInterface.cpp

```diff
@@ -39,7 +39,11 @@ int LoWPANNDInterface::connect()
     // Release mutex before blocking
     nanostack_unlock();
 
-    int32_t count = connect_semaphore.wait(30000);
+    // wait connection for ever:
+    // -routers will create new network and get local connectivity
+    // -end devices will get connectivity once attached to existing network
+    // -devices without network settings gets connectivity once commissioned and attached to network
```

Comment:
Comments are applicable to thread. 6lowpan would need different wording about routers, host etc. But I think that there is no need to specify here what happens i.e. leaving only comment "wait connection for ever" is enough.